waku/v2/node/waku_setup.nim

```diff
@@ -0,0 +1,112 @@
+{.push raises: [Defect].}
+
+## Collection of utilities commonly used
+## during the setup phase of a Waku v2 node
+
+import
+  std/tables,
+  chronos,
+  chronicles,
+  json_rpc/rpcserver,
+  metrics,
+  metrics/chronos_httpserver,
+  stew/results,
+  stew/shims/net,
+  ./storage/sqlite,
+  ./storage/migration/migration_types,
+  ./jsonrpc/[admin_api,
+             debug_api,
+             filter_api,
+             relay_api,
+             store_api,
+             private_api,
+             debug_api],
+  ./config,
+  ./wakunode2
+
+logScope:
+  topics = "wakunode.setup"
+
+type
+  SetupResult*[T] = Result[T, string]
+
+##########################
+# Setup helper functions #
+##########################
+
+proc startRpc*(node: WakuNode, rpcIp: ValidIpAddress, rpcPort: Port, conf: WakuNodeConf)
+  {.raises: [Defect, RpcBindError, CatchableError].} =
+  # @TODO: API handlers still raise CatchableError
+
+  let
+    ta = initTAddress(rpcIp, rpcPort)
+    rpcServer = newRpcHttpServer([ta])
+  installDebugApiHandlers(node, rpcServer)
+
+  # Install enabled API handlers:
+  if conf.relay:
+    let topicCache = newTable[string, seq[WakuMessage]]()
+    installRelayApiHandlers(node, rpcServer, topicCache)
+    if conf.rpcPrivate:
+      # Private API access allows WakuRelay functionality that 
+      # is backwards compatible with Waku v1.
+      installPrivateApiHandlers(node, rpcServer, node.rng, topicCache)
+  
+  if conf.filter:
+    let messageCache = newTable[ContentTopic, seq[WakuMessage]]()
+    installFilterApiHandlers(node, rpcServer, messageCache)
+  
+  if conf.store:
+    installStoreApiHandlers(node, rpcServer)
+  
+  if conf.rpcAdmin:
+    installAdminApiHandlers(node, rpcServer)
+  
+  rpcServer.start()
+  info "RPC Server started", ta
+
+proc startMetricsServer*(serverIp: ValidIpAddress, serverPort: Port) {.raises: [Defect, Exception].} =
+    info "Starting metrics HTTP server", serverIp, serverPort
+    
+    startMetricsHttpServer($serverIp, serverPort)
```

Comment:
It doesn't matter that much but I guess it is cleaner to try/except the `Exception` here and raiseAssert in case it is not a Defect

Reply:
Thanks! Done in 80b71d1